src/app/mail/mail-detail/mail-detail.component.ts

```diff
@@ -141,12 +141,101 @@ export class MailDetailComponent implements OnInit, OnDestroy {
     this.router.navigateByUrl(`/mail/${this.mailFolder}`);
   }
 
-  onPrint() {
+  onPrint(): void {
     if (this.decryptedContent) {
-      const printWindow = window.open();
-      printWindow.document.write(this.decryptedContent);
-      printWindow.print();
-      printWindow.close();
-    }
+
+        let popupWin;
+
+        const subject = document.getElementById('mail-subject').innerHTML;
```

Comment:
@zainzafar90 can't we get all this info directly from `this.mail`?

Reply:
it extracts the markup not just the data, that's why I used innerHTML